drivers/hmis/app/graphql/mutations/update_external_form_submission.rb

```diff
@@ -41,8 +41,8 @@ def resolve(id:, input:, project_id: nil)
             error_out(record.enrollment.client.errors.full_messages) unless record.enrollment.client.valid?
             error_out(record.enrollment.errors.full_messages) unless record.enrollment.valid?
 
-            record.enrollment.client.save!
             record.enrollment.save_new_enrollment!
```

Comment:
Save the enrollment first; otherwise the enrollment gets saved alongside the client as a related record, but then `save_new_enrollment` raises because it's not a new enrollment. (The reason to use `save_new_enrollment` is to respect the project's auto enter config)

Reply:
hrm, you are right to 👀 , something is still wonky here. This does not work when auto enter is turned on in the project; the client has to already exist when save_new_enrollment is called.

Reply:
Ah, so it's specifically an issue when geolocation is collected. Since location is related to both client and enrollment, when client is saved first, location is saved, and enrollment is persisted too. Then the 'Unexpected: save_new_enrollment called on a persisted enrollment' error is thrown when we try to save the new enrollment.

Ideas for workarounds:
- Does `save_new_enrollment!` _need_ to raise if the enrollment is already persisted? I feel a bit hesitant to change this when it may have far-reaching consequences, but grepping in the code it does not actually look that widely used. 
- On the other hand: Setting the location's `source` to the HMIS enrollment before it has been persisted is already proving tricky for us, so maybe this is one more prompting in the direction of post-processing location after everybody else has already been persisted (instead of keeping the hack in `after_save`)

Reply:
Discussed on a call, we will keep `save_new_enrollment` as is, since in general we only want to call this method on a brand new enrollment. We'll separately call similar code `save_and_auto_enter` from the external forms mutation keeping the impact localized.